lib/cudadrv/synchronization.jl

```diff
@@ -133,15 +134,25 @@ function synchronization_worker(data)
 end
 
 @noinline function create_synchronization_worker(i)
-    sync_channels[i] = BidirectionalChannel{SyncObject,CUresult}()
-    # should be safe to assign before threads are running;
-    #  any user will just submit work that makes it block
+    lock(sync_channel_lock) do
+        # test and test-and-set
```

Comment:
TATAS doesn't look safe here to me, since `BidirectionalChannel` is a struct, not a pointer, so it doesn't have atomic updates in an array. Maybe make it mutable for now? This is the reason for https://github.com/JuliaLang/julia/pull/51495, but we don't have that merged yet. It could also now be an AtomicMemory object, but (a) that is only in v1.11 (b) it doesn't have the public API written yet (c) would incur a lock allocation for every element and every read or write access, which seems undesired here